math/wxMaxima/Portfile

```diff
@@ -1,59 +1,109 @@
+    fetch.type  git
+    git.url     https://github.com/andrejv/wxmaxima
+    git.branch  b32fed0d535c05d589d1f573d2c0f4f839180d18
+
+    post-fetch {
+        system -W ${worksrcpath} "git describe --tags"
+    }
 
-    set docPath ${prefix}/share/doc/${name}
-    xinstall -m 755 -d ${destroot}${docPath}
-    foreach f { AUTHORS COPYING README } {
-        xinstall -m 644 ${worksrcpath}/${f} ${destroot}${docPath}
+    patchfiles      patch-devel-src_Dirstructure.cpp.diff
+    post-patch {
+        reinplace -W ${worksrcpath} "s|/OPT/LOCAL|${prefix}|g" \
+                                    src/Dirstructure.cpp
     }
+
+    livecheck.type  none
+} else {
+    # get the source tarball from github.
+    PortGroup       github 1.0
+
+    github.setup    andrejv wxmaxima 18.02.0 Version-
+    checksums       rmd160  d1017d56bfd76e9c51dcc3230be89ebca1724560 \
+                    sha256  77f0414be54d8ea15acedfc3cedf72a57cc2b3ba5d4835a53f3009e050c05483 \
+                    size    10131293
+
+    patchfiles      patch-src_Dirstructure.cpp.diff
+    post-patch {
+        reinplace -W ${worksrcpath} "s|/OPT/LOCAL|${prefix}|g" \
+                                    src/Dirstructure.cpp
+    }
+    
+    livecheck.regex Version-(\[a-z0-9.\]+)${extract.suffix}
 }
 
+universal_variant   no
+
+variant wx32 description {Use wxWidgets 3.2 - experimental!} {}
+
+if {[variant_exists wx32] && [variant_isset wx32]} {
+    # we want to use the wxWidgets-3.2
+    wxWidgets.use   wxWidgets-3.2
+} else {
+    wxWidgets.use   wxWidgets-3.0
+}
+
+depends_lib         port:${wxWidgets.port}
+
+depends_build       port:cmake
+
+depends_run         port:maxima
+
+patch.pre_args      -Np0
+
+use_configure       no
+
+build {
+    system -W ${worksrcpath}       "mkdir ./build"
+    system -W ${worksrcpath}/build "cmake -DwxWidgets_CONFIG_EXECUTABLE:FILEPATH=${wxWidgets.wxconfig} .."
+    system -W ${worksrcpath}/build "cmake --build ."
+}
+
+destroot {
+    xinstall -m 755 -d ${destroot}${applications_dir}
+    file copy ${worksrcpath}/build/wxMaxima.app ${destroot}${applications_dir}
+}
 
+notes "
+    Be careful: \n\
+    If you want to use wxMaxima with default setting, \n\
+    you should enter the command \n\
+    \"open -a ${applications_dir}/wxMaxima.app\" \n\
+    on Terminal.app in order to launch wxMaxima. \n\
+    Because you must let wxMaxima know the path to softwares \n\
+    installed with MacPorts. \n\n\
+    On the other hand, \n\
+    when you launch directly wxMaxima on the Finder, \n\
+    the path to maxima (e.g. \"${prefix}/bin/maxima\") is automatically set \n\
+    on the configuration dialog. \n\
+    But you should enter the maxima expression \n\
+    gnuplot_command:\"${prefix}/bin/gnuplot\"\$ \n\
+    in order to use gnuplot. \n\n\
+    Some configuration files are automatically created by wxMaxima. \n\
+    You can enter the command \n\
+    \"find \$HOME/Library -name '*wxMaxima*'\" \n\
+    on Terminal.app in order to find them.
+"
```

Comment:
I believe it should be possible to have this handled automatically in the .app bundle. Someone with more experience might want to pipe up.

Reply:
If the info is not necessary, I will remove it.

Reply:
Oh, the information would be necessary if nothing is done to fix the PATH for the app.